### ISSUE TITLE
Cannot save borderless polygon in vector editor

### ISSUE THREAD

**Issue by AndersBillLinden** [LLK/scratch-flash#746]
When I draw a polygon with the vector editor, I can fill the edge with the empty color (the color in the palette that is a white square with diagonal red line). In the editor, it looks like a borderless polygon, but in the stage, it will still show up as a polygon with edges.

![image](https://cloud.githubusercontent.com/assets/1017466/7461814/0c9ed0da-f2ad-11e4-9067-f3c2a3297dd3.png)

It will also save as a polygon with an edge, even if I save it locally as a sb2 file, so next time I open the project, the polygon has edges in the editor as well.

![image](https://cloud.githubusercontent.com/assets/1017466/7461819/1960b748-f2ad-11e4-849b-2b0208d57e7b.png)

If I open the file 1.svg inside the zip archive being a sb2 file, it will have two path elements.

``` xml
<svg version="1.1" width="104" height="91" viewBox="-1 -1 104 91" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <!-- Exported by Scratch - http://scratch.mit.edu/ -->
  <path id="ID0.1292742001824081" fill="#995900" stroke="#000000" stroke-width="2" stroke-linecap="round" d="M 187 99.500 L 185.250 124 L 274.500 186.500 L 282.500 184.500 L 286 161 L 200 100.500 L 187 99.500 Z " transform="matrix(1, 0, 0, 1, -184, -98)"/>
  <path id="ID0.2674711593426764" fill="#663B00" stroke="#000000" stroke-width="2" stroke-linecap="round" d="M 185.800 123.650 L 187.750 100.500 L 276.750 164 L 274.800 185.300 L 185.800 123.650 Z " transform="matrix(1, 0, 0, 1, -184, -98)"/>
</svg>
```

and they will both apparently have the attributes `stroke="#000000" stroke-width="2"`.

If I alter the svg file using a text editor, removing the edge of the second polygon, it will open in scratch having the intended appearance.

![image](https://cloud.githubusercontent.com/assets/1017466/7461853/5e809fb4-f2ad-11e4-974f-109218ffdc44.png)

**Comment by sclements** [comment]
It sounds like we have a bug in the SVG export code. This is probably a small amount of work to track down and fix.

**Comment by TheLogFather** [comment]
@sclements: isn't this just something that's put in place to prevent making an invisible line? Note that the only place where a transparent-filled line becomes (temporarily) invisible is in the costume editor itself - the sprite never 'sees' it, and switching to another costume and back makes it re-appear in the costume editor.
Is it something similar to line 98 of PathTool.as that's causing it...?

**Comment by Paddle2See** [comment]
@sclements - I don't think it's just in the SVG export code as you can see this problem without exporting anything.  Just make a vector circle, fill the edge with some color other than transparent (it updates in the editor and on the stage), then attempt to fill the edge with transparent (it updates in the editor - but not on the stage).  Forcing a save doesn't help - the edge can not be made transparent once it has been filled with a non-transparent color.

**Comment by TheLogFather** [comment]
Yup, that sums up what I was getting at above. It's something that's been a tad annoying for a while - that the only way to make some borderless shape, such as a no-outline rect, is to choose it to be so at the start (i.e. so it's selected as a filled rect rather than outlined rect).

If you try to change an outlined rect to have a transparent outline, the editor won't allow it (though at first it appears to in the costume editor). Furthermore, if you start with a filled, non-outlined rect and accidentally fill its outline, you can't get back to having no outline.

It looks to me like it's probably some code that's preventing the user from making a line invisible, along similar lines to https://github.com/LLK/scratch-flash/blob/master/src/svgeditor/tools/PathTool.as#L98

One telling procedure to follow is this:
1) Start from a blank vector costume
2) Click the rect tool
3) Choose a filled rect (rather than outline)
4) Choose a colour that's clearly distinct from black (e.g. green)
5) Create the rect
You now have a solid rect, filled with some colour, and with no outline. Continue:
6) Switch to the fill tool
7) Switch the colour to transparent
8) Now, carefully, move the fill cursor to where the rect outline would be
9) Try to fill the already-invisible outline with transparent
10) Note that the rect now has a black outline as shown on the stage, but *not* in costume editor
11) Switch to the stage, then back to the sprite
12) Note that the rect now also has black outline in the costume editor
In other words, the fill was successful, but it 'refused' to actually use transparent to fill it.

**Comment by AndersBillLinden** [comment]
@TheLogFather How do you specify filled rect in 3) ?

**Comment by AndersBillLinden** [comment]
I close this issue report and make some new ones of it.

**Comment by 2jour** [comment]
I am going to reopen this .. @AndersBillLinden for now

**Comment by AndersBillLinden** [comment]
The thing is that I was importing the image from svg. I did not use scratch to draw the polygon. There isnt even a polygon tool in scratch. So I remembered it wrong when writing the issue. The fllling of the edge however was done in scratch, so it is an actual issue. I also think the issue could be simplified a lot.
